test/e2e/integration/003-multi-block-selection.js

```diff
@@ -0,0 +1,61 @@
+describe( 'Multi-block selection', () => {
+	before( () => {
+		cy.newPost();
+	} );
+
+	it( 'Should select/unselect multiple blocks', () => {
+
+		const lastBlockSelector = '.editor-visual-editor__block-edit:last [contenteditable="true"]:first';
+		const firstBlockContainerSelector = '.editor-visual-editor__block:first';
+		const lastBlockContainerSelector = '.editor-visual-editor__block:last';
+		const multiSelectedCssClass = 'is-multi-selected';
+
+		// Creating test blocks
+		// Using the placeholder
+		cy.get( '[value="Write your story"]' ).click();
+		cy.get( lastBlockSelector ).type( 'First Paragraph' );
+
+		// Using the quick inserter
+		cy.get( '.editor-visual-editor__inserter [aria-label="Insert Paragraph"]' ).click();
+		cy.get( lastBlockSelector ).type( 'Second Paragraph' );
+
+
+		// Default: No selection
+		cy.get( firstBlockContainerSelector ).should('not.have.class', multiSelectedCssClass);
+		cy.get( lastBlockContainerSelector ).should('not.have.class', multiSelectedCssClass);
+
+
+		// Multiselect via Shift + click
+		cy.get( firstBlockContainerSelector ).click();
+		cy.get('body').type( '{shift}', { release: false } );
+		cy.get( lastBlockContainerSelector ).click();
+
+		// Verify selection
+		cy.get( firstBlockContainerSelector ).should('have.class', multiSelectedCssClass);
+		cy.get( lastBlockContainerSelector ).should('have.class', multiSelectedCssClass);
+
+		// Unselect
+		cy.get('body').type( '{shift}' ); // releasing shift
+		cy.get( lastBlockContainerSelector ).click();
+
+		// No selection
+		cy.get( firstBlockContainerSelector ).should('not.have.class', multiSelectedCssClass);
+		cy.get( lastBlockContainerSelector ).should('not.have.class', multiSelectedCssClass);
+
+
+		// Multiselect via keyboard
+		cy.get('body').type( '{ctrl}a' );
```

Comment:
This is failing locally, because on MacOS the shortcut is `{meta}a`. A simple way to fix this is to type both (and explain with a comment). But I wonder if there's a nicer way to fix it.